wsd/DocumentBroker.cpp

```diff
@@ -1553,27 +1553,26 @@ bool DocumentBroker::updateStorageLockState(const Authorization& auth, std::stri
             return true;
             break;
         case StorageBase::LockUpdateResult::UNAUTHORIZED:
-            LOG_ERR("Failed to " << "Locked docKey [" << _docKey
-                                 << "]. Invalid or expired access token");
+            LOG_ERR("Failed to lock docKey [" << _docKey << "]. Invalid or expired access token");
             break;
         case StorageBase::LockUpdateResult::FAILED:
-            LOG_ERR("Failed to " << "Locked docKey [" << _docKey << "] with reason [" << error
-                                 << ']');
+            LOG_ERR("Failed to lock docKey [" << _docKey << "] with reason: " << error);
             break;
     }
 
     return false;
 }
 
-bool DocumentBroker::updateStorageLockState(ClientSession& session, bool lock, std::string& error)
+bool DocumentBroker::updateStorageLockState(ClientSession& session, StorageBase::LockState lock,
+                                            std::string& error)
 {
     if (session.getAuthorization().isExpired())
     {
         error = "Expired authorization token";
         return false;
     }
 
-    if (lock && session.isReadOnly())
+    if (lock == StorageBase::LockState::LOCK && session.isReadOnly())
```

Comment:
`lock` could be `const`, yes (perhaps it should). But I'm not a fan of reverse-comparisons. It's far more natural to ask "is X zero?", "is Y empty?", "is Z equal to 5?" rather than "is 0 X?", "is empty Y?", and "is 5 equal to Z?"

I know some prefer the reverse approach, because it protects against accidentally assigning instead of comparing (i.e. missing a second '='), but modern compilers warn hard on this, and past the first few years of experience most people learn not to do that mistake anyway. But the cost to the reader (and we read far more than we write) to read the const first, think "I'm comparing 0 to something..." then see the variable think about its type and meaning, only to remember what constant we were comparing it with, well, it's overkill. (Also, in many cases we compare variables to variables. What to do in those cases?)